runtime-sdk/src/modules/rofl/mod.rs

```diff
@@ -124,22 +141,130 @@ impl<Cfg: Config> Module<Cfg> {
 
         // Set genesis parameters.
         Self::set_params(genesis.parameters);
+
+        // Insert all applications.
+        for cfg in genesis.apps {
+            state::set_app(cfg);
+        }
+    }
+
+    /// Create a new ROFL application.
+    #[handler(call = "rofl.Create")]
+    fn tx_create<C: Context>(ctx: &C, body: types::Create) -> Result<app_id::AppId, Error> {
+        <C::Runtime as Runtime>::Core::use_tx_gas(Cfg::GAS_COST_CALL_CREATE)?;
+
+        if CurrentState::with_env(|env| env.is_check_only()) {
+            return Ok(Default::default());
+        }
+
+        let (creator, tx_index) =
+            CurrentState::with_env(|env| (env.tx_caller_address(), env.tx_index()));
+        let app_id = app_id::AppId::from_creator_round_index(
+            creator,
+            ctx.runtime_header().round,
+            tx_index.try_into().map_err(|_| Error::InvalidArgument)?,
+        );
+
+        // Sanity check that the application doesn't already exist.
+        if state::get_app(app_id).is_some() {
+            return Err(Error::AppAlreadyExists);
+        }
+
+        // Transfer stake.
+        <C::Runtime as Runtime>::Accounts::transfer(
+            creator,
+            *ADDRESS_APP_STAKE_POOL,
+            &Cfg::STAKE_APP_CREATE,
+        )?;
+
+        // Register the application.
+        let cfg = types::AppConfig {
+            id: app_id,
+            policy: body.policy,
+            admin: Some(creator),
+            stake: Cfg::STAKE_APP_CREATE,
+        };
+        state::set_app(cfg);
+
+        CurrentState::with(|state| state.emit_event(Event::AppCreated { id: app_id }));
+
+        Ok(app_id)
+    }
+
+    /// Ensure caller is the current administrator, return an error otherwise.
+    fn ensure_caller_is_admin(cfg: &types::AppConfig) -> Result<(), Error> {
+        let caller = CurrentState::with_env(|env| env.tx_caller_address());
+        if cfg.admin != Some(caller) {
+            return Err(Error::Forbidden);
+        }
+        Ok(())
+    }
+
+    /// Update a ROFL application.
+    #[handler(call = "rofl.Update")]
+    fn tx_update<C: Context>(ctx: &C, body: types::Update) -> Result<(), Error> {
+        <C::Runtime as Runtime>::Core::use_tx_gas(Cfg::GAS_COST_CALL_UPDATE)?;
+
+        let mut cfg = state::get_app(body.id).ok_or(Error::UnknownApp)?;
+
+        // Ensure caller is the admin and is allowed to update the configuration.
+        Self::ensure_caller_is_admin(&cfg)?;
+
+        if CurrentState::with_env(|env| env.is_check_only()) {
+            return Ok(());
+        }
+
+        cfg.policy = body.policy;
+        cfg.admin = body.admin;
```

Comment:
Talked offline, and loosing/transferring stake to another admin is fine and efficient.

I would maybe check that `cfg.admin != body.admin || cfg.policy != body.policy` so that you don't get events when nothing changes.